src/twisted/web/http_headers.py

```diff
@@ -168,7 +184,39 @@ def removeHeader(self, name: AnyStr) -> None:
 
         @return: L{None}
         """
-        self._rawHeaders.pop(self._encodeName(name), None)
+        self._rawHeaders.pop(_encodeName(name), None)
+
+    def _setRawHeadersFaster(self, name: _EncodedHeader, values: List[bytes]) -> None:
+        """
+        Like C{setRawHeaders}, but faster.
+
+        @param name: The name of the HTTP header to set the values for.
+
+        @param values: A list of bytes, each one being a header value of the
+            given name. These will be still be sanitized since otherwise it's
+            too easy for security bugs to allow unescaped whitespace through.
+        """
+        self._rawHeaders[name] = [_sanitizeLinearWhitespace(v) for v in values]
+
+    def _getRawHeaderLastFaster(self, name: _EncodedHeader) -> Optional[bytes]:
+        """
+        Get the last header value, if any.
+
+        @param name: The name of the HTTP header to get the value for.
+        """
+        result = self._rawHeaders.get(name, None)
+        if result is not None:
+            return result[-1]
+        else:
+            return None
+
+    def _getRawHeadersFaster(self, name: _EncodedHeader) -> Optional[List[bytes]]:
```

Comment:
I think that this should have been the default behaviour for `getRawHeaders` 

Can we have this implemnted as `__getitem__` ?

so that you can do something like `request.requestHeaders.get(name, None)`

Since this is the `Headers` class, maybe we can call it just `getRaw` ... the `headers` part of `getRawHeaders` is kind of redundand

and it's needs bytes as input... and you are on your own for requesting the canonical header name.

```suggestion
    def getRawValues(self, name: bytes) -> Optional[List[bytes]]:
```